code/game/machinery/computer/camera.dm

```diff
@@ -142,6 +136,70 @@ var/global/list/tv_monitors = list()
 
 	light_color = LIGHT_COLOR_PINK
 
+/obj/machinery/computer/security/proc/set_camera(var/mob/living/user, var/obj/machinery/camera/C)
+	if(C)
+		if ((!Adjacent(user) || user.machine != src || user.blinded || user.isStunned() || !( C.can_use() )) && (!istype(user, /mob/living/silicon/ai)))
+			if(!C.can_use() && !isAI(user))
+				src.current = null
+			user.cancel_camera()
+			return 0
+		else
+			if(isAI(user))
+				var/mob/living/silicon/ai/A = user
+				A.eyeobj.forceMove(get_turf(C))
+				A.client.eye = A.eyeobj
+			else
+				src.current = C
+				use_power(50)
+
+	user.set_machine(src)
+	user.reset_view(current)
+
+/obj/machinery/computer/security/proc/next(var/mob/living/user)
+
+	var/list/net = cameranet.camera_list()
+
+	var/place = net.Find(current)
+	var/place_0 = place
+	var/found = FALSE
+	var/seen_it_all = FALSE
```

Comment:
For some reason I can't explain, `next` does increase the index by 1, but `net[n + 1] == net[n]`  in _all_ cases I've been testing
Further more, `net[n - 1] =! net[n]`
I'm assuming that there is some duplication in the list and that `place` finds the first entry each time, no L[n + 1] = L[n] but L[n - 1] =! L[n]

I have no idea *why* this is happening nor how to reliably test it.